### DISCUSSION TITLE
how do you read this json file

### DISCUSSION THREAD

**Discussion by djouallah** [duckdb/duckdb#10676] [Q&A] [answered]
I just can't figure out how to read it
[SCADA_2023-12-15.json](https://github.com/duckdb/duckdb/files/14291658/SCADA_2023-12-15.json)

**Comment by Mause** [comment]
Can you please upload the file again? Looks like you clicked create on the discussion before it actually uploaded. A sample of the error would be useful as well

**Reply by djouallah** [reply]
no error, I don't know how to parse it

**Reply by Mause** [reply]
It seems to load fine for me?

```text
⚫◗ select data from 'SCADA_2023-12-15.json';                     
┌────────────────────────────────────────────────────────────────┐
│                                     data                                                                                                                          │
│ struct(facilityscadadispatchintervals struct(dispatchinterval varchar, cod…                                                 │
├────────────────────────────────────────────────────────────────┤
│ {'facilityScadaDispatchIntervals': [{'dispatchInterval': 2023-12-15T08:00:…                                                 │
└────────────────────────────────────────────────────────────────┘
```

**Reply by djouallah** [reply]
yes but how do you convert it to a proper table

**Reply by Mause** [reply]
```sql
CREATE TABLE whatever AS FROM 'SCADA_2023-12-15.json';
```

etc?

**Reply by djouallah** [reply]
sorry probably lost in translation, how I can get dispatch Interval and other columns  into a tabular format not json.

**Reply by Mause** [reply]
If you want to access fields of structs, you use the `.` operator.

```sql
select data.facilityScadaDispatchIntervals from 'SCADA_2023-12-15.json';
```

which will give you an array, which you can unnest using the `unnest` function

```sql
select unnest(data.facilityScadaDispatchIntervals) from 'SCADA_2023-12-15.json';
```

In the future, the more information you can give up front, the better

**Reply by djouallah** [reply]
thanks but it does not work
<img width="704" alt="image" src="https://github.com/duckdb/duckdb/assets/12554469/11e54a1c-adfb-4457-a846-89c8eb0d5a4b">

**Reply by Mause** [reply, marked as answer]
Again, the more information you provide up front, the better we can help you.

If you want to explode a struct into columns, you do `data.*`:

```sql
select data.*
from (select unnest(data.facilityScadaDispatchIntervals) as data from 'SCADA_2023-12-15.json');
```

**Reply by djouallah** [reply]
Thank you
